roxie/ccd/ccdactivities.cpp

```diff
@@ -5301,14 +5301,14 @@ class CRoxieDummyActivityFactory : public CSlaveActivityFactory  // not a real a
                 if (indexName && !allFilesDynamic)
                 {
                     bool isOpt = pretendAllOpt || _graphNode.getPropBool("att[@name='_isIndexOpt']/@value");
-                    indexfile.setown(_queryFactory.queryPackage().lookupFileName(indexName, isOpt, true, true, _queryFactory.queryWorkUnit()));
+                    indexfile.setown(_queryFactory.queryPackage().lookupFileName(indexName, isOpt, true, true, _queryFactory.queryWorkUnit(), !queryFactory.isDynamic()));
```

Comment:
Except these two - here we should be passing true but we should also have the queryFactory.allFilesDynamic() test on the if condition above.